packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -3,36 +3,52 @@ import { Spinner } from '@podman-desktop/ui-svelte';
 
 type SearchFunction = (s: string) => Promise<string[]>;
 
-// the text displayed when no option is selected
-export let placeholder: string | undefined = undefined;
-export let required: boolean = false;
-export let delay: number = 200;
-export let disabled: boolean = false;
-export let initialFocus: boolean = false;
-export let id: string | undefined = undefined;
-export let name: string | undefined = undefined;
-export let error: boolean = false;
-
-export let searchFunction: SearchFunction = async (_s: string) => [];
-export let onChange = function (_s: string) {};
-export let onEnter = function () {};
-
-let input: HTMLInputElement;
-let list: HTMLDivElement;
-let scrollElements: HTMLElement[] = [];
-let value: string;
-let items: string[] = [];
-let inputDelayTimeout: NodeJS.Timeout;
-let opened: boolean = false;
-let highlightIndex: number = -1;
-let pageStep = 10;
-let userValue: string = '';
-let loading: boolean = false;
+interface Props {
+  placeholder?: string;
+  required?: boolean;
+  delay?: number;
+  disabled?: boolean;
+  initialFocus?: boolean;
+  id?: string;
+  name?: string;
+  error?: boolean;
+  searchFunction?: SearchFunction;
+  onChange?: (value: string) => void;
+  onEnter?: () => void;
+  class?: string;
+}
+
+let {
+  placeholder,
+  required = false,
+  delay = 200,
+  disabled = false,
+  initialFocus = false,
+  id,
+  name,
+  error = false,
+  searchFunction = async (_s: string) => [],
+  onChange = function (_s: string) {},
+  onEnter = function () {},
```

Comment:
This is just a migration, so I kept the default values, but this should not be kept, a follow-up will arrive to remove those.